### PR TITLE
fix: 修复切换页面 changeConfirmLocale 被清空问题

### PR DESCRIPTION
**Pull request by crazyair**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |    Fix ConfigProvider within multiple page switch will lose locale issue.      |
| 🇨🇳 Chinese |    修复多页面使用 ConfigProvider 在切换时，locale 会丢失的问题。      |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/locale-provider/index.tsx [hunk 1]
```diff
@@ -62,6 +62,10 @@ export default class LocaleProvider extends React.Component<LocaleProviderProps,
     );
   }
 
+  componentDidMount() {
+    changeConfirmLocale(this.props.locale && this.props.locale.Modal);
```
Comment (zombieJ) [inline components/locale-provider/index.tsx:66]: 放这里是不是构造函数里的就可以去掉了？

Comment (crazyair) [reply components/locale-provider/index.tsx:66]: 看了单元测试
image
有的会再 DidMount 使用，所以不能去，现在 DidMount 加上，是为了修复 router 切换导致丢失问题

**Comment by crazyair** [comment]
问题说明
image

A 切换 B 生命周期是正常的，但是 B 再切回 A，就会先执行 A 的 WillMount 再执行 B 的 WillUnMount ，导致 locale 被清空

**Comment by crazyair** [comment]
[Edit on CodeSandbox](https://codesandbox.io/s/anniuleixing-antd4123-forked-s2viv?file=/example.js)

**Comment by zombieJ** [comment]
👌 来个测试用例吧

**Comment by crazyair** [comment]
@afc163 怎么不执行 pretest 了？

**Pull request merged**